cpp/libs/gst_nmos_audio_receiver_plugin/utils.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+#include <string>
+#include <glib.h>
+#include <variant>
+#include <gst/gst.h>
+#include <nlohmann/json.hpp>
+
+typedef struct node_fields_t
+{
+    std::string id;
+    std::string configuration_location;
+} node_fields_t;
+
+typedef struct device_fields_t
+{
+    std::string id;
+    std::string label;
+    std::string description;
+} device_fields_t;
+
+typedef struct config_fields_t
+{
+    node_fields_t node;
+    device_fields_t device;
+    std::string id;
+    std::string label;
+    std::string description;
+    std::string address;
+    gint port;
+    std::string interface_name;
+} config_fields_t;
+
+template <typename T = GstElement> class GstElementHandle
+{
+  public:
+    static std::variant<GstElementHandle, std::nullptr_t> create_element(const char* factory_name,
+                                                                         const char* element_name = nullptr)
+    {
+        T* elem = reinterpret_cast<T*>(gst_element_factory_make(factory_name, element_name));
+        if(!elem)
+        {
+            return nullptr;
+        }
+        return GstElementHandle(elem);
+    }
+
+    static std::variant<GstElementHandle, std::nullptr_t> create_bin(const char* bin_name)
+    {
+        T* bin = reinterpret_cast<T*>(gst_bin_new(bin_name));
+        if(!bin)
+        {
+            return nullptr;
+        }
+        return GstElementHandle(bin);
+    }
+
+    GstElementHandle(const GstElementHandle&)            = delete;
+    GstElementHandle& operator=(const GstElementHandle&) = delete;
+
+    // Move constructor
+    GstElementHandle(GstElementHandle&& other) noexcept : handle_(other.handle_) { other.handle_ = nullptr; }
+
+    // Move assignment
+    GstElementHandle& operator=(GstElementHandle&& other) noexcept
+    {
+        if(this != &other)
+        {
+            if(handle_ != nullptr)
+            {
+                gst_object_unref(handle_);
+            }
+            handle_       = other.handle_;
+            other.handle_ = nullptr;
+        }
+        return *this;
+    }
+
+    ~GstElementHandle()
+    {
+        if(handle_)
+        {
+            gst_object_unref(handle_);
+        }
+    }
+
+    void reset(T* new_ptr = nullptr) { handle_ = new_ptr; }
```

Comment:
We need a good comment here. The semantics are not obvious at all. What happens to the owned handle, if any? What happens to the new handle?

Reply:
If the goal is to use always pass nullptr as the parameter, so that this no longer keeps track of the handle and won't call unref, then let's rather define a function that only does that and that is aptly named. E.g.

```
/// Destruction of this object will no longer unref the owned element.
void forget() 
{
    handle_ = new_ptr;
}
```